### PR TITLE
Add docs for setting node config in E2E tests

### PR DESCRIPTION
**Pull request by lukaszcl**

### CODE REVIEW

File: integration-tests/testconfig/README.md [hunk 1]
```diff
@@ -116,6 +116,83 @@ When processing TOML files, the system initially searches for a general (unnamed
 
 Finally `default.toml` file is envisioned to contain fundamental and universally applicable settings, such as logging configurations.
 
+### Chainlink Node TOML config
+
+To set custom config for Chainlink Node use `NodeConfig.BaseConfigTOML` in TOML. Example:
+
+```
```
Comment (kalverra) [inline integration-tests/testconfig/README.md]:
```suggestion
```toml
```

File: integration-tests/testconfig/README.md [hunk 2]
```diff
@@ -116,6 +116,83 @@ When processing TOML files, the system initially searches for a general (unnamed
...
+MaxSize = '0b'
+
+[OCR]
+Enabled = true
+DefaultTransactionQueueDepth = 0
+"""
+```
+
+To set base config for EVM network use `NodeConfig.CommonChainConfigTOML`. Example:
+```
```
Comment (kalverra) [inline integration-tests/testconfig/README.md]:
```suggestion
```toml
```

File: integration-tests/testconfig/README.md [hunk 3]
```diff
@@ -116,6 +116,83 @@ When processing TOML files, the system initially searches for a general (unnamed
...
+
+[GasEstimator]
+PriceMax = '200 gwei'
+LimitDefault = 6000000
+FeeCapDefault = '200 gwei'
+"""
+```
+
+To set custom per-chain config use `[NodeConfig.ChainConfigTOMLByChainID]`. Example:
+```
```
Comment (kalverra) [inline integration-tests/testconfig/README.md]:
```suggestion
```toml
```

File: integration-tests/testconfig/README.md [hunk 4]
```diff
@@ -116,6 +116,83 @@ When processing TOML files, the system initially searches for a general (unnamed
...
+"""
+```
+
+For more examples see `example.toml` in product TOML configs like `testconfig/automation/example.toml`.
+
+### Setting env vars for Chainlink Node
+
+To set env vars for Chainlink Node use `WithCLNodeOptions()` and `WithNodeEnvVars()` when building a test environment. Example:
+
+```
```
Comment (kalverra) [inline integration-tests/testconfig/README.md]:
```suggestion
```go
```

File: integration-tests/testconfig/README.md [hunk 5]
```diff
@@ -116,6 +116,83 @@ When processing TOML files, the system initially searches for a general (unnamed
 
 Finally `default.toml` file is envisioned to contain fundamental and universally applicable settings, such as logging configurations.
 
+### Chainlink Node TOML config
+
+To set custom config for Chainlink Node use `NodeConfig.BaseConfigTOML` in TOML. Example:
```
Comment (Tofel) [inline integration-tests/testconfig/README.md]: I think we need richer description. That would tell at least:
* where is the default node config
* how to override it
* mention that you cannot override just a single value, e.g. `FeedsManager = true` but whole config

File: integration-tests/testconfig/README.md [hunk 6]
```diff
@@ -116,6 +116,83 @@ When processing TOML files, the system initially searches for a general (unnamed
...
+[Log.File]
+MaxSize = '0b'
+
+[OCR]
+Enabled = true
+DefaultTransactionQueueDepth = 0
+"""
+```
+
+To set base config for EVM network use `NodeConfig.CommonChainConfigTOML`. Example:
```
Comment (Tofel) [inline integration-tests/testconfig/README.md]: Same here and I would say that this is the default chain-specific config that is used, when no custom one matches selected network

File: integration-tests/testconfig/README.md (hunk 5 above)
Comment (lukaszcl) [reply integration-tests/testconfig/README.md]: @Tofel isn't default and overrides described in `Configuration and Overrides`? Or this is something else?

File: integration-tests/testconfig/README.md (hunk 2 above)
Comment (lukaszcl) [reply integration-tests/testconfig/README.md]: fixed in 04cd351581d6d3c23f9d9959e8854f969d4e4083

File: integration-tests/testconfig/README.md (hunk 4 above)
Comment (lukaszcl) [reply integration-tests/testconfig/README.md]: fixed in 04cd351581d6d3c23f9d9959e8854f969d4e4083

File: integration-tests/testconfig/README.md (hunk 3 above)
Comment (lukaszcl) [reply integration-tests/testconfig/README.md]: fixed in 04cd351581d6d3c23f9d9959e8854f969d4e4083

File: integration-tests/testconfig/README.md (hunk 1 above)
Comment (lukaszcl) [reply integration-tests/testconfig/README.md]: fixed in 04cd351581d6d3c23f9d9959e8854f969d4e4083

File: integration-tests/testconfig/README.md (hunk 5 above)
Comment (lukaszcl) [reply integration-tests/testconfig/README.md]: @Tofel pls check 04cd351581d6d3c23f9d9959e8854f969d4e4083

File: integration-tests/testconfig/README.md (hunk 6 above)
Comment (lukaszcl) [reply integration-tests/testconfig/README.md]: @Tofel pls check 04cd351581d6d3c23f9d9959e8854f969d4e4083

**Review by Tofel — APPROVED** [review]

**Review by kalverra — APPROVED** [review]

**Pull request merged**